### PR TITLE
feat: Improve React, introduce TypeScript rules

### PR DESCRIPTION
**Pull request by markkkkas**
Resolves: https://edlioinc.atlassian.net/browse/CMSDV-617

### CODE REVIEW

**Review by ErnestasZaloga — APPROVED** [review]
🚀

**Review by ghost — APPROVED** [review]

**Review by ia64mail — COMMENTED** [review]
Let's invite Core team to review since these changes are Edlio global. 

@markkkkas Would it affect projects other than CMS? What about CMS itself?

**Comment by markkkkas** [comment]
@ia64mail I wanted first to gather reviews & feedback from our team, then invite core team. It will affect only react apps, since i only edited react rules.

**Comment by lancedolan** [comment]
Looks good, guys; keep 'em coming 🤙

File: react.js [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
-    'react/jsx-first-prop-new-line': ['error', 'multiline'],
-    'react/sort-comp': 'error'
-  }
-}
+    "react/display-name": "off",
+    "react/no-unused-prop-types": "error",
+    "react/no-unused-state": "error",
+    "react/jsx-first-prop-new-line": ["error", "multiline"],
+    "react/sort-comp": "error",
+    semi: "error",
```
Comment (SlavaBakshin) [inline react.js:24]: The same question https://github.com/edlio/cms/pull/14233#pullrequestreview-1762178787

**Review by SlavaBakshin — APPROVED** [review]

**Review by SlavaBakshin — APPROVED** [review]

**Review by raulingg — APPROVED** [review]

**Review by AliLezamaIgrat — APPROVED** [review]

**Review by Hmorales94 — APPROVED** [review]

**Review by ia64mail — APPROVED** [review]

**Pull request merged**